### PR TITLE
[Auto Parallel] Adapt Partitioner & DistOp for ERNIE3.0 Inference and cache

### PR DESCRIPTION
**Pull request by JZ-LIANG**
### PR types
Function optimization

### PR changes
Others

### Describe
自动并行适配 ERNIE 预测中的 while 和cache 机制

### CODE REVIEW

File: python/paddle/distributed/auto_parallel/operators/dist_fill_constant_batch_size_like.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        dist_op_context = ctx.dist_op_context
+        src_op = dist_op_context.cur_src_op
+        op_dist_attr = ctx.get_op_dist_attr_for_program(src_op)
+        main_block = dist_op_context.work_block
+        op = main_block.ops[-1]
+        assert op.type == "fill_constant_batch_size_like"
+
+        # modify shape attr according to how output are partitioned
+        out_name = op.output('Out')[0]
+        dim_mapping = op_dist_attr.get_output_dims_mapping(out_name)
```
Comment (aoyulong) [inline python/paddle/distributed/auto_parallel/operators/dist_fill_constant_batch_size_like.py]: Please rename `dim_mapping` to `dims_mapping`.

File: python/paddle/distributed/auto_parallel/operators/dist_fill_constant_batch_size_like.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        out_name = op_desc.output('Out')[0]
+        out_dims_mapping = op_dist_attr.get_output_dims_mapping(out_name)
+        shape_list = op_desc.attr("shape")
+
+        if len(shape_list) != len(out_dims_mapping):
+            return False
+
+        return True
+
+    def is_auto_compatible(self, dist_op):
```
Comment (aoyulong) [inline python/paddle/distributed/auto_parallel/operators/dist_fill_constant_batch_size_like.py:64]: The compatible restrictions may be a little loose.

File: python/paddle/distributed/auto_parallel/operators/dist_fill_constant_batch_size_like.py (hunk 1 above)
Comment (JZ-LIANG) [reply python/paddle/distributed/auto_parallel/operators/dist_fill_constant_batch_size_like.py]: fixed

File: python/paddle/distributed/auto_parallel/operators/dist_fill_constant_batch_size_like.py (hunk 2 above)
Comment (JZ-LIANG) [reply python/paddle/distributed/auto_parallel/operators/dist_fill_constant_batch_size_like.py:64]: fixed

**Review by aoyulong — APPROVED** [review]
LGTM

**Pull request merged**